longshot/application/main.tf

```diff
@@ -0,0 +1,218 @@
+  name = "/mandrill/api-key"
+}
+
+resource "heroku_app" "app" {
+  name   = "dosomething-${var.name}"
+  region = "us"
+
+  organization {
+    name = "dosomething"
+  }
+
+  config_vars {
+    # App settings:
+    APP_ENV   = "production"
+    APP_DEBUG = "false"
+    APP_LOG   = "errorlog"
+    APP_URL   = "https://${var.host}"
+
+    # Drivers:
+    QUEUE_DRIVER   = "sqs"
+    CACHE_DRIVER   = "redis"
+    SESSION_DRIVER = "redis"
+    STORAGE_DRIVER = "s3"
+    MAIL_DRIVER    = "mandrill"
+
+    # Email:
+    EMAIL_NAME      = "${var.email_name}"
+    EMAIL_ADDRESS   = "${var.email_address}"
+    MAIL_HOST       = "smtp.mandrillapp.com"
+    MANDRILL_APIKEY = "${data.aws_ssm_parameter.mandrill_api_key.value}"
+
+    # Database:
+    DB_HOST     = "${aws_db_instance.database.address}"
+    DB_PORT     = "${aws_db_instance.database.port}"
+    DB_DATABASE = "${var.database_name}"
+    DB_USERNAME = "${data.aws_ssm_parameter.database_username.value}"
+    DB_PASSWORD = "${data.aws_ssm_parameter.database_password.value}"
+
+    # S3 Bucket & SQS Queue:
+    AWS_ACCESS_KEY    = "${aws_iam_access_key.aws_key.id}"
+    AWS_SECRET_KEY    = "${aws_iam_access_key.aws_key.secret}"
+    SQS_DEFAULT_QUEUE = "${aws_sqs_queue.queue.id}"
+    S3_REGION         = "${aws_s3_bucket.storage.region}"
+    S3_BUCKET         = "${aws_s3_bucket.storage.id}"
+
+    # Additional secrets, set manually:
+    # APP_KEY = ...
+  }
+
+  buildpacks = [
+    "heroku/nodejs",
+    "heroku/php",
+  ]
+
+  acm = true
+}
+
+resource "heroku_formation" "web" {
+  app      = "${heroku_app.app.name}"
+  type     = "web"
+  size     = "${var.web_size}"
+  quantity = "${var.web_scale}"
+}
+
+resource "heroku_formation" "queue" {
+  app      = "${heroku_app.app.name}"
+  type     = "queue"
+  size     = "${var.queue_size}"
+  quantity = "${var.queue_scale}"
+}
+
+resource "heroku_domain" "domain" {
+  app      = "${heroku_app.app.name}"
+  hostname = "${var.host}"
+}
+
+resource "heroku_drain" "papertrail" {
+  app = "${heroku_app.app.name}"
+  url = "syslog+tls://${var.papertrail_destination}"
+}
+
+resource "heroku_pipeline_coupling" "app" {
+  app      = "${heroku_app.app.name}"
+  pipeline = "${var.pipeline}"
+  stage    = "${var.pipeline_stage}"
+}
+
+resource "heroku_addon" "redis" {
+  app  = "${heroku_app.app.name}"
+  plan = "heroku-redis:${var.redis_type}"
+}
+
+resource "aws_db_instance" "database" {
+  engine              = "mariadb"
+  engine_version      = "10.0"
+  instance_class      = "${var.database_type}"
+  allocated_storage   = "${var.database_size}"
+  username            = "${data.aws_ssm_parameter.database_username.value}"
+  password            = "${data.aws_ssm_parameter.database_password.value}"
+  publicly_accessible = true
+  skip_final_snapshot = true
+
+  tags = {
+    Application = "${var.name}"
+  }
+}
+
+resource "aws_sqs_queue" "queue" {
+  name                      = "${var.name}"
+  message_retention_seconds = "${60 * 60 * 24 * 14}"
```

Comment:
if there are constants in tf, might make sense to define the constant in the beginning of the file.

Reply:
Yep, Terraform has [local values](https://www.terraform.io/docs/configuration/locals.html) which act as constants. Since this value won't be used elsewhere, I'm going to just add a comment to this line to better explain it's usage in #55.